packages/vitest/src/utils/graph.ts

```diff
@@ -29,7 +29,18 @@ export async function getModuleGraph(ctx: Vitest, id: string): Promise<ModuleGra
     graph[id] = (await Promise.all(mods.map(m => get(m, seen)))).filter(Boolean) as string[]
     return id
   }
-  await get(ctx.server.moduleGraph.getModuleById(id))
+
+  const workspaceProject = ctx.projects.find(project => project.getName() === workspaceName)
+  const workspaceModules = workspaceProject?.ctx.getModuleProjects(id)
+  const selectedModule = workspaceModules?.find(module => module.getName() === workspaceName)
+  let workspaceMod: ModuleNode | undefined
+  if (selectedModule) {
+    const { server, browser } = selectedModule
+    const mod = server.moduleGraph.getModuleById(id) || browser?.moduleGraph.getModuleById(id)
+    workspaceMod = mod
+  }
+
+  await get(ctx.server.moduleGraph.getModuleById(id) || workspaceMod)
```

Comment:
Shouldn't you fallback to `ctx.server.moduleGraph` only if you didn't find the `workspaceMod`? This line looks suspicious to me